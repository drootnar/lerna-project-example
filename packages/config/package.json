{
  "name": "@lerna-pro/config",
  "version": "0.0.1",
  "description": "lerna-pro config",
  "main": "lib/src/index.js",
  "scripts": {
    "clean": "rimraf lib",
    "test": "jest --logHeapUsage",
    "test-watch": "jest --watch --logHeapUsage",
    "build": "webpack --progress --color"
  },
  "dependencies": {
    "babel-core": "^6.26.3",
    "babel-preset-env": "^1.7.0",
    "babel-preset-stage-2": "^6.24.1"
  }
}
