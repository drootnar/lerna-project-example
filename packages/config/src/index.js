import logLevel from './logLevel'

export default {
  logLevel
}
