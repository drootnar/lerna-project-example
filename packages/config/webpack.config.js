const path = require('path');

const files = [
  'index.js',
  'logLevel.js'
]

function generateWebpackConfig() {
  let config = []
  files.forEach(function(file) {
    config.push({
      entry: './src/'+file,
      output: {
        filename: file,
        path: path.resolve(__dirname, 'lib')
      }
    });
  });
  return config;
}

module.exports = generateWebpackConfig();
