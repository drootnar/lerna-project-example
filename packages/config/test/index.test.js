import config from '../src'

describe('config', () => {
  it('returns dict with config values', async () => {
    expect(config).toMatchSnapshot()
  })
})
