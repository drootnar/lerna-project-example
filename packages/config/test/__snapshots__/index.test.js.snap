// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config returns dict with config values 1`] = `
Object {
  "logLevel": "debug",
}
`;
